### PR TITLE
Update apply-setters from v0.1 to v0.2

### PR DESCRIPTION
**Pull request by adamdoes**
Now that #546 has been merged, we should update `set-project-id` to use the new version of `apply-setters` so that the following repro steps won't produce an error:

```sh
# Get the empty blueprint
kpt pkg get https://github.com/GoogleCloudPlatform/blueprints@v0.3.0
cd blueprints/catalog/empty

# Run set-project-id
kpt fn eval ./ --image gcr.io/kpt-fn/set-project-id:v0.1.0 --include-meta-resources -- 'project-id=some-value'

# Run apply-setters
kpt fn render ./
```

### CODE REVIEW

File: functions/go/set-project-id/main.go
Comment (mengqiy) [reply functions/go/set-project-id/main.go:111]: > I think it would be better for us(maintainers) to do that. Let's unblock this PR.

Can you please take care of it?

Comment (adamdoes) [reply functions/go/set-project-id/main.go:111]: (ended up taking Thursday/Friday off, so I didn't get a chance to work on this until today)

I updated my PR to incorporate the changes everywhere. [This](https://github.com/GoogleContainerTools/kpt-functions-catalog/blob/9e34d68f4927b0be741535920856c588d0f786f2/functions/go/fix/fixpkg/fix.go#L284) is the only code change that isn't just a find/replace.

The E2E tests aren't passing, but they aren't passing on the base repo for me either, so perhaps I've configured them wrong.

File: examples/set-project-id-advanced/.expected/diff.patch
Comment (mengqiy) [inline examples/set-project-id-advanced/.expected/diff.patch:29]: Looking at the output from the CI:
```
        Diff of Diff
        ===
        --- /tmp/kpt-e2e-diff-087313666/expected	2021-08-30 20:36:14.544060022 +0000
        +++ /tmp/kpt-e2e-diff-087313666/actual	2021-08-30 20:36:14.544060022 +0000
        @@ -20,7 +20,7 @@
            storage-class: standard
         +  project-id: foo
         diff --git a/subpkg/Kptfile b/subpkg/Kptfile
        -index c96b2cd..2643351 100644
        +index c96b2cd..52b8640 100644
         --- a/subpkg/Kptfile
         +++ b/subpkg/Kptfile
         @@ -2,3 +2,8 @@ apiVersion: kpt.dev/v1
```

Line 23 `index c96b2cd..2643351 100644` needs to be updated to `index c96b2cd..52b8640 100644`.

Comment (adamdoes) [reply examples/set-project-id-advanced/.expected/diff.patch:29]: Updated it! 👍

**Comment by mengqiy** [comment]
`examples/fix-simple` is also failing:
```
        Diff of Diff
        ===
        --- /tmp/kpt-e2e-diff-747059285/expected	2021-08-30 22:18:28.555486855 +0000
        +++ /tmp/kpt-e2e-diff-747059285/actual	2021-08-30 22:18:28.555486855 +0000
        @@ -1,5 +1,5 @@
         diff --git a/Kptfile b/Kptfile
        -index a8e7f27..a10b4df 100644
        +index a8e7f27..30e5250 100644
         --- a/Kptfile
         +++ b/Kptfile
         @@ -1,20 +1,24 @@
```
Similar change is needed.

**Comment by mengqiy** [comment]
We have an ongoing issue with `CheckSite`. It's not blocking for merging the PR.

**Comment by adamdoes** [comment]
> `examples/fix-simple` is also failing:

Ack; [fixed](https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/550/commits/3e429053fca2f6eda6970eab56b78a633ef4ceda).

> We have an ongoing issue with `CheckSite`. It's not blocking for merging the PR.

Ahh, I see. I think this PR should be all set, then. 👍

Let me know if there's anything else I should do!

**Review by mengqiy — APPROVED** [review]

File: functions/go/set-project-id/main.go
Comment (mengqiy) [reply functions/go/set-project-id/main.go:111]: > > I think it would be better for us(maintainers) to do that. Let's unblock this PR.
> 
> Can you please take care of it?

Sorry about the confusion here. I was asking @phanimarupaka if he can help.

**Pull request merged**
